java/arcs/core/entity/CollectionHandle.kt

```diff
@@ -56,20 +56,23 @@ class CollectionHandle<T : Entity>(
 
     override suspend fun isEmpty() = fetchAll().isEmpty()
 
-    override suspend fun fetchAll() = adaptValues(storageProxy.getParticleView())
+    override suspend fun fetchAll() = checkNotClosed {
+        adaptValues(storageProxy.getParticleView())
+    }
     // endregion
 
     // region implement QueryCollectionHandle<T, Any>
-    override suspend fun query(args: Any): Set<T> =
+    override suspend fun query(args: Any): Set<T> = checkNotClosed {
```

Comment:
I wonder if it'd make more sense to call this `checkPreconditions` or something?  This way it could be expanded to include the check that the current thread is the expected thread when we get to the single-threaded particle world.